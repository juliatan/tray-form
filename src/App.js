import React from 'react';
import Layout from './hoc/Layout/Layout';

function App() {
  return (
    <Layout>
      <p>Tray.io</p>
    </Layout>
  );
}

export default App;
