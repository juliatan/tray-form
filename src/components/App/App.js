import React from 'react';
import Layout from '../../hoc/Layout/Layout';

function App() {
  return (
    <Layout>
      <h1>Tray.io</h1>
    </Layout>
  );
}

export default App;
