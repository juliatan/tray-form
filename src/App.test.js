import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  afterEach(cleanup);

  it('should render', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });

  it('should render header text', () => {
    const { getByText } = render(<App />);
    const h1Element = getByText(/Tray.io/i);
    expect(h1Element).toBeInTheDocument();
  });
});
