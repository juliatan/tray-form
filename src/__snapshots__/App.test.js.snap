// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render 1`] = `
<DocumentFragment>
  <div
    class="Layout"
  >
    <h1>
      Tray.io
    </h1>
    <nav>
      <ol
        class="FormTabs"
      >
        <li
          class="FormTab  active"
        >
          <a
            class="Link"
            href="#"
          >
            User
          </a>
        </li>
        <li
          class="FormTab middle "
        >
          <a
            class="Link"
            href="#"
          >
            Privacy
          </a>
        </li>
        <li
          class="FormTab  "
        >
          <a
            class="Link"
            href="#"
          >
            Done
          </a>
        </li>
      </ol>
    </nav>
    <div
      class="Page"
    >
      <div
        class="UserDetails"
      >
        <p
          class="Legend"
        >
          * is a required field
        </p>
        <form
          action="#"
        >
          <div
            class="LabelContainer"
          >
            <label
              for="name"
            >
              Name: 
              <span
                class="Required"
              >
                *
              </span>
            </label>
          </div>
          <input
            name="name"
            placeholder="Jane Doe"
            type="text"
            value=""
          />
          <div
            class="LabelContainer"
          >
            <label
              for="role"
            >
              Role:
            </label>
          </div>
          <input
            name="role"
            placeholder="Engineer"
            type="text"
            value=""
          />
          <div
            class="LabelContainer"
          >
            <label
              for="email"
            >
              Email: 
              <span
                class="Required"
              >
                *
              </span>
            </label>
          </div>
          <input
            name="email"
            placeholder="name@company.com"
            type="email"
            value=""
          />
          <div
            class="LabelContainer"
          >
            <label
              for="password"
            >
              Password: 
              <span
                class="Required"
              >
                *
              </span>
            </label>
          </div>
          <input
            name="password"
            placeholder="Min. 10 chars"
            type="password"
            value=""
          />
          <button
            class="Button"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
